vault/login_mfa.go

```diff
@@ -647,6 +674,87 @@ func (b *LoginMFABackend) loginMFAMethodExistenceCheck(eConfig *mfa.MFAEnforceme
 	return aggErr.ErrorOrNil()
 }
 
+// sanitizeMethodNameSameNamespace returns the name without the namespace.
+// The method name could contain the namespace path.
+// If the namespace is specified in the name, check if the namespace
+// matches the request namespace. If so, trim the namespace path and
+// return the name. If it does not match the request namespace,
+// return an error. If the namespace path is not included in the name,
+// nothing happens.
+func (b *LoginMFABackend) sanitizeMethodNameSameNamespace(ns *namespace.Namespace, name string) (string, error) {
+	if name == "" {
+		return "", nil
+	}
+
+	if ns == nil {
+		return "", fmt.Errorf("namespace cannot be nil")
+	}
+
+	if !strings.Contains(name, "/") {
+		return name, nil
+	}
```

Comment:
If the goal of this method is to return the method name without the embedded namespace, should this check exist above the `ns == nil` check?